### DISCUSSION TITLE
How to set auth method in docker?

### DISCUSSION THREAD

**Discussion by theAlevan** [filebrowser/filebrowser#1966] [Q&A]
I just can't find any documentation how to set my preferred auth method while running docker contaner.

I found references for making changes with .filebrowser.json file, and I tried using this found on a github comment to disable auth:

```
{
  "port": 80,
  "baseURL": "",
  "address": "",
  "log": "/filebrowser.log",
  "database": "/database.db",
  "root": "/srv",
  "noauth": true
}
```

But it's not working. Are there any other ways?

**Comment by ramiresviana** [comment]
Users can access Filebrowser without authentication when the `noauth` option is used, it only works when the database was not initialized yet. The authentication method can only be set using the `config set` command. You can configure your container to run commands before starting the Filebrowser or use a existing database file with your configuration.

**Reply by ingrove** [reply]
I'm sorry, I don't follow your comments.  I have a docker installation that I want to run with noauth authentication.  I don't understand what you mean by "it only works when the database was not initialized yet."  How is it possible to select the noauth option if the application isn't running and the database hasn't been initialized?  Could you provide an example to illustrate how to configure my container to run the necessary commands prior to starting the Filebrowser?

**Reply by theAlevan** [reply]
I embedded this line in my docker compose:
`command: "config set --auth.method=noauth"`

**Reply by ramiresviana** [reply]
Options are used in command line, configuration files and environment variables allowing you to change Filebrowser behavior. Check https://filebrowser.org/cli/filebrowser

**Reply by ingrove** [reply]
Thank you for your reply.  I was able to set the authentication to noauth by including FB_NOAUTH=true in my docker run command.
